alf/utils/common.py

```diff
@@ -625,11 +625,18 @@ def write_gin_configs(root_dir, gin_file):
 def warning_once(msg, *args):
     """Generate warning message once.
 
+    Note that the current implementation resembles that of the ``log_every_n()```
+    function in ``logging`` but reduces the calling stack by one to ensure
+    the multiple warning once messages generated at difference places can be
+    displayed correctly.
```

Comment:
Does this mean that log_every_n has a bug?

Reply:
Currently  log_every_n can only get the caller one level up, and its count of n is based on this retrieved caller.